src/test/java/org/folio/search/controller/SortAuthorityIT.java

```diff
@@ -0,0 +1,126 @@
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+
+import java.util.Collections;
+import java.util.stream.IntStream;
+import org.folio.cql2pgjson.model.CqlSort;
+import org.folio.search.domain.dto.Authority;
+import org.folio.search.support.base.BaseIntegrationTest;
+import org.folio.search.utils.types.IntegrationTest;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+@IntegrationTest
+class SortAuthorityIT extends BaseIntegrationTest {
+
+  @BeforeAll
+  static void prepare() {
+    setUpTenant(4, authorities());
+  }
+
+  @AfterAll
+  static void cleanUp() {
+    removeTenant();
+  }
+
+  @Test
+  void canSortAuthoritiesByHeadingType_asc() throws Exception {
+    doSearchByAuthorities(allAuthoritiesSortedBy("headingType", ASCENDING))
+      .andExpect(jsonPath("totalRecords", is(4)))
+      .andExpect(jsonPath("authorities[0].headingType", is("Corporate Name")))
+      .andExpect(jsonPath("authorities[1].headingType", is("Other")))
+      .andExpect(jsonPath("authorities[2].headingType", is("Personal Name")))
+      .andExpect(jsonPath("authorities[3].headingType", is("Uniform Title")));
+  }
+
+  @Test
+  void canSortAuthoritiesByHeadingType_desc() throws Exception {
+    doSearchByAuthorities(allAuthoritiesSortedBy("headingType", DESCENDING))
+      .andExpect(jsonPath("totalRecords", is(4)))
+      .andExpect(jsonPath("authorities[0].headingType", is("Uniform Title")))
+      .andExpect(jsonPath("authorities[1].headingType", is("Personal Name")))
+      .andExpect(jsonPath("authorities[2].headingType", is("Other")))
+      .andExpect(jsonPath("authorities[3].headingType", is("Corporate Name")));
+  }
+
+  @Test
+  void canSortAuthoritiesByHeadingRef_asc() throws Exception {
+    doSearchByAuthorities(allAuthoritiesSortedBy("headingRef", ASCENDING))
+      .andExpect(jsonPath("totalRecords", is(4)))
+      .andExpect(jsonPath("authorities[0].headingRef", is("111")))
+      .andExpect(jsonPath("authorities[1].headingRef", is("aaa")))
+      .andExpect(jsonPath("authorities[2].headingRef", is("bbb")))
+      .andExpect(jsonPath("authorities[3].headingRef", is("ccc")));
+  }
+
+  @Test
+  void canSortAuthoritiesByHeadingRef_desc() throws Exception {
+    doSearchByAuthorities(allAuthoritiesSortedBy("headingRef", DESCENDING))
+      .andExpect(jsonPath("totalRecords", is(4)))
+      .andExpect(jsonPath("authorities[0].headingRef", is("ccc")))
+      .andExpect(jsonPath("authorities[1].headingRef", is("bbb")))
+      .andExpect(jsonPath("authorities[2].headingRef", is("aaa")))
+      .andExpect(jsonPath("authorities[3].headingRef", is("111")));
+  }
+
+  @Test
+  void canSortAuthoritiesByAuthRefType_asc() throws Exception {
+    doSearchByAuthorities(allAuthoritiesSortedBy("authRefType", ASCENDING))
+      .andExpect(jsonPath("totalRecords", is(4)))
+      .andExpect(jsonPath("authorities[0].authRefType", is("Auth/Ref")))
```

Comment:
add validation that secondary sorting works well too for `authRefType` and `headingType`